File: app/src/main/java/com/example/client/api/service/TransactionService.java
package com.example.client.api.service;

import com.example.client.models.message.MessageModel;
import com.example.client.models.transaction.TransactionModel;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface TransactionService {
    @FormUrlEncoded
    @POST("views/transaction/insert.php")
    Call<MessageModel> insert(@Field("user") int user,
                              @Field("subject") int subject,
                              @Field("amount") Double amount);

    @GET("views/transaction/checkInsert.php")
    Call<MessageModel> checkInsert(@Query("user") int user,@Query("subject") int subject);

    @GET("views/transaction/getByUser.php")
    Call<List<TransactionModel>> getByUser(@Query("user") int user);

    @GET("views/transaction/getByOrderCode.php")
    Call<TransactionModel> getByOrderCode(@Query("ordercode") String ordercode);

}

File: app/src/main/java/com/example/client/api/service/TransactionService.kt
package com.example.client.api.service

import com.example.client.models.payment.MomoRequest
import com.example.client.models.response.BaseResponse
import com.example.client.models.transaction.DataTransactionResponse
import com.example.client.models.transaction.TransactionResponse
import io.reactivex.Observable
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface TransactionService {
    @POST("api/transaction/recharge_create.php")
    fun createRecharge(@Body momoRequest: MomoRequest): Observable<BaseResponse<DataTransactionResponse>>

    @GET("api/transaction/transaction_getAll.php")
    fun getTransactions(
            @Query("user_id") user_id: Int,
            @Query("type") type: String,
            @Query("page") page: Int,
            @Query("limit") limit: Int
    ): Observable<BaseResponse<List<TransactionResponse>>>

}